gtda/utils/validation.py

```diff
@@ -154,3 +156,151 @@ def validate_params(parameters, references, exclude=None):
     parameters_ = {key: value for key, value in parameters.items()
                    if key not in exclude_}
     return _validate_params(parameters_, references)
+
+
+def check_list_of_images(X, **kwargs):
+    """Check a list of arrays representing images, by iterating through
+    the input one by one. To pass a test when `kwargs` is empty,
+    all images ``x``, ``y`` in `X` must satisfy:
+        - ``x.ndim >= 2``,
+        - ``all(np.isfinite(x))``,
+        - ``x.shape == y.shape``.
+
+    Parameters
+    ----------
+    X : list of ndarray
+        Each entry of `X` corresponds to an image.
+
+    kwargs : dict, optional, default: ``{}``
```

Comment:
I'm sorry. I should have thought it better before suggesting ``{}``. In reality I don't think this is good either as the user would not actually pass a dictionary here, so it is misleading. I think we should adopt an approach as in https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html, i.e.
```
**kwargs
        Keyword arguments accepted by
        :func:`~gtda.utils.validation.check_list_of_arrays`.
```